File: binarysearch.py
def binary(arr, low, high, x):

	if high >= low:

		mid = (high + low) // 2

		
		if arr[mid] == x:
			return mid

		elif arr[mid] > x:
			return binary(arr, low, mid - 1, x)

		else:
			return binary(arr, mid + 1, high, x)

	else:
		return -1
		
arr = [ 2, 3, 4, 8, 45 ]
x = 45

result = binary(arr, 0, len(arr)-1, x)

if result != -1:
	print("Element is present at index", str(result))
else:
	print("Element is not present in array")

File: binarysearch.cpp
#include <bits/stdc++.h>
using namespace std;

int binary(int arr[], int l, int r, int x)
{
	if (r >= l) {
		int mid = l + (r - l) / 2;


		if (arr[mid] == x)
			return mid;


		if (arr[mid] > x)
			return binary(arr, l, mid - 1, x);


		return binary(arr, mid + 1, r, x);
	}

	return -1;
}

int main(void)
{
	int arr[] = { 2, 3, 1, 4, 5 };
	int x = 10;
	int n = sizeof(arr) / sizeof(arr[0]);
	int result = binary(arr, 0, n - 1, x);
	(result == -1)
		? cout << "Element is not present in array"
		: cout << "Element is present at index " << result;
	return 0;
}
